ext/libclementine-tagreader/tagreader.cpp

```diff
@@ -502,7 +501,12 @@ void TagReader::ReadFile(const QString& filename,
           song->set_score(score);
         }
       }
-
+      if (items.contains("tmpo")) {
+	float bpm = (float)items["tmpo"].toInt();
+	if (song->bpm() <= 0 && bpm > 0) {
```

Comment:
Could `has_bpm()` be used here?

Reply:
tmpo and BPM are just translations. BPM origins from id3v2 , tmpo is the the equivalent apple/MP4 tagname. I dont think that translation can be used in member name (has_bpm()). As novice I cant answer that.

Reply:
`song` is a protobuf object and bpm is an optional field, so `song->has_bpm()` would tell you if it has been previously set. But, again, if the bpm may have previously been set to 0, then this looks right.

Reply:
The change worked for me, but as I said: Only to get BPM. For tagging I use kid3, ffmpeg, AtomicParsley, ... and all I wanted was to see BPM in clementine. For dance music BPM is essential and when amarok died I found clementine as a natural successor, Thank you for making me aware of "protobuf", "optional", ... I did not look into song  class. I only compared id3 to mp4 handling in tagreader and made a suggestion, the fix in mainstream code should be done by you as you are quite more familiar with that code.

Reply:
I don't know, the code looks alright to me. I guess the lint checker doesn't run in /ext because it looks like there are some tabbing issue.